### PR TITLE
CCM-2575: Add 404 tests for dev

### PR DESCRIPTION
**Pull request by Kennedy1265**
## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Checklist
* [ ] Brief description of work completed, and any technical decisions made as part of the PR
* [ ] PR link added as a comment to the relevant JIRA ticket
* [ ] Branch deployed to a dynamic env - link to deployment pipeline
* [ ] PR link shared on Slack and/or Teams
* [ ] 1 reviews received

### CODE REVIEW

File: docs/tests/development/get_v1_messages/validation.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Validation Tests
+
+These tests target the API endpoint GET /v1/messages testing 404 responses when invalid data is provided.
+
+
```
Comment (PhilipSkinner) [inline docs/tests/development/get_v1_messages/validation.md:5]: Looks like there should be some content in here @Kennedy1265?

File: tests/lib/assertions.py [hunk 2]
```diff
@@ -211,7 +211,8 @@ def assert_error_with_optional_correlation_id(resp, code, error, correlation_id)
             else:
                 assert error in response_errors
 
-        assert resp.headers.get("X-Correlation-Id") == correlation_id
+        if correlation_id is not None:
```
Comment (PhilipSkinner) [inline tests/lib/assertions.py:214]: Does this need to be here? When we've passed `None` in as the correlation id in the past this hasn't caused us any issues

File: docs/tests/development/get_v1_messages/validation.md (hunk 1 above)
Comment (Kennedy1265) [reply docs/tests/development/get_v1_messages/validation.md]: Updated - also renamed to be in line with sandbox

File: tests/lib/assertions.py (hunk 2 above)
Comment (Kennedy1265) [reply tests/lib/assertions.py]: Removed

**Review by PhilipSkinner — APPROVED** [review]
Approved

**Pull request merged**